src/shared/serverinfo.hpp

```diff
@@ -0,0 +1,231 @@
+#ifndef SERVERINFO_HPP
+#define SERVERINFO_HPP
+
+#include <vector>
+#include <string>
+
+#include "defs.hpp"
+
+struct serverinfo;
+
+// list of known servers, with or without informations
+// TODO:
+//   * should be a private static of serverinfo
+//   * stop using raw pointers (RAII ftw)
+extern vector<serverinfo*> servers;
+
+// code imported, here to reduce build dependencies
```

Comment:
These are not "imports", they're forward declarations. Is that what you meant? Or do I miss something here?

There should be a TODO to remove those again before merging. They might've helped speed up your development, however they increase coupling between code entities a lot.